ibis/pyspark/compiler.py

```diff
@@ -1060,65 +1033,27 @@ def compile_percent_rank(t, expr, scope, *, window, **kwargs):
 @compiles(ops.NTile)
 def compile_ntile(t, expr, scope, *, window, **kwargs):
     op = expr.op()
-
     buckets = op.buckets.op().value
     return F.ntile(buckets).over(window)
 
 
 @compiles(ops.FirstValue)
 def compile_first_value(t, expr, scope, *, window, **kwargs):
     op = expr.op()
-
     src_column = t.translate(op.arg, scope)
     return F.first(src_column).over(window)
 
 
 @compiles(ops.LastValue)
 def compile_last_value(t, expr, scope, *, window, **kwargs):
     op = expr.op()
-
     src_column = t.translate(op.arg, scope)
     return F.last(src_column).over(window)
 
 
 @compiles(ops.RowNumber)
 def compile_row_number(t, expr, scope, *, window, **kwargs):
-    return F.row_number().over(window).astype('long') - F.lit(1)
-
-
-def _handle_cumulative_operation(t, expr, scope, fn, *, window, **kwargs):
-    op = expr.op()
-
-    src_column = t.translate(op.arg, scope)
-    return fn(src_column).over(window)
-
-
-@compiles(ops.CumulativeSum)
-def compile_cumulative_sum(t, expr, scope, *, window, **kwargs):
-    return _handle_cumulative_operation(
-        t, expr, scope, F.sum, window=window, **kwargs
-    )
-
-
-@compiles(ops.CumulativeMean)
-def compile_cumulative_mean(t, expr, scope, *, window, **kwargs):
-    return _handle_cumulative_operation(
-        t, expr, scope, F.mean, window=window, **kwargs
-    )
-
-
-@compiles(ops.CumulativeMin)
-def compile_cumulative_min(t, expr, scope, *, window, **kwargs):
-    return _handle_cumulative_operation(
-        t, expr, scope, F.min, window=window, **kwargs
-    )
-
-
-@compiles(ops.CumulativeMax)
-def compile_cumulative_max(t, expr, scope, *, window, **kwargs):
-    return _handle_cumulative_operation(
-        t, expr, scope, F.max, window=window, **kwargs
-    )
+    return F.row_number().over(window).cast('long') - 1
```

Comment:
@icexelloss could you provide a docstring here using numpy format? https://numpydoc.readthedocs.io/en/latest/format.html